src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/URLIndexInterface.java

```diff
@@ -0,0 +1,15 @@
+package au.org.emii.geoserver.wms;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public interface URLIndexInterface {
+    // If timstamp is null, returns last URL sorted by time
+    public String getUrlForTimestamp(LayerDescriptor layerDescriptor, String timestamp) throws IOException;
+
+    public List<String> getTimesForDay(LayerDescriptor layerDescriptor, String day) throws IOException;
+
+    public Map<Integer, Map<Integer, Set<Integer>>> getUniqueDates(LayerDescriptor layerDescriptor) throws IOException;
```

Comment:
Interface methods shouldn't have access modifiers (e.g. public). It's a public interface so every method must eb public.